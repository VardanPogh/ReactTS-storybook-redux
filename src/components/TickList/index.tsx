import React from 'react'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(() => ({
  container: {
    display: 'flex',
    alignItems: 'center',
    '& img': {
      marginRight: '32px',
    },
  },
  title: {
    color: '#474747',
    fontSize: '24px',
    marginBottom: '15px',
    marginLeft: '64px',
    textAlign: 'left',
  },
}))

interface itemsElements {
  id: string
  label: string
}

interface Props {
  items: Array<itemsElements>
  title: string
  img: string
}

const TickList: React.FC<Props> = ({ items, title, img }) => {
  const classes = useStyles()
  const List = items.map((item: itemsElements, key: number) => {
    return (
      <div className={classes.container} key={key}>
        <img src={img} alt="icon" />
        <p>{item.label}</p>
      </div>
    )
  })

  return (
    <>
      <p className={classes.title}>{title}</p>
      {List}
    </>
  )
}

export default TickList
