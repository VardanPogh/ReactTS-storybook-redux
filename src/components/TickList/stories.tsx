import React from 'react'
import { withKnobs, object, text } from '@storybook/addon-knobs'

import TickList from '.'

export default {
  title: 'Tick',
  component: TickList,
  decorators: [withKnobs],
}
const List = [
  {
    id: 'label-1',
    label: 'Provide requirements over long meetings',
  },
  {
    id: 'label-2',
    label: 'Delivers in weeks or months',
  },
  {
    id: 'label-3',
    label: 'Revision in days or weeks',
  },
]

export const Default: React.FC = () => (
  <TickList
    items={object('List', List)}
    title={text('Title', 'Consultants')}
    img={text('Img', 'img/check.png')}
  />
)
