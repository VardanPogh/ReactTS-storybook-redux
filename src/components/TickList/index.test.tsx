import React from 'react'
import { render } from '@testing-library/react'
import TickList from '.'

const List = [
  {
    id: 'label-1',
    label: 'Provide requirements over long meetings',
  },
  {
    id: 'label-2',
    label: 'Delivers in weeks or months',
  },
  {
    id: 'label-3',
    label: 'Revision in days or weeks',
  },
]

describe('RadioButton Test', () => {
  test('Loads RadioButton', () => {
    const { container } = render(
      <TickList items={List} title="Consultants" img="img/check.png" />
    )
    expect(container).toMatchSnapshot()
  })
})
