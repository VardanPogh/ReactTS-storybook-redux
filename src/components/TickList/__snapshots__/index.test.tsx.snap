// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Test Loads RadioButton 1`] = `
<div>
  <p
    class="makeStyles-title-2"
  >
    Consultants
  </p>
  <div
    class="makeStyles-container-1"
  >
    <img
      alt="icon"
      src="img/check.png"
    />
    <p>
      Provide requirements over long meetings
    </p>
  </div>
  <div
    class="makeStyles-container-1"
  >
    <img
      alt="icon"
      src="img/check.png"
    />
    <p>
      Delivers in weeks or months
    </p>
  </div>
  <div
    class="makeStyles-container-1"
  >
    <img
      alt="icon"
      src="img/check.png"
    />
    <p>
      Revision in days or weeks
    </p>
  </div>
</div>
`;
